src/misc/paper-write.ts

```diff
@@ -0,0 +1,244 @@
+import { Material, Sound } from 'org.bukkit';
+import {
+  EquipmentSlot,
+  ItemStack,
+  PlayerInventory,
+} from 'org.bukkit.inventory';
+import { PlayerInteractEvent } from 'org.bukkit.event.player';
+import { Player } from 'org.bukkit.entity';
+import { color, text } from 'craftjs-plugin/chat';
+import { ChatColor } from 'net.md_5.bungee.api';
+import { isRightClick } from '../common/helpers/click';
+import { ChatMessage, GLOBAL_PIPELINE } from '../chat/pipeline';
+import { CustomItem } from '../common/items/CustomItem';
+import { VkItem } from '../common/items/VkItem';
+import * as yup from 'yup';
+import { isBuffer } from 'lodash';
+
+// Max length of text.
+const MAX_LENGTH = 128;
+
+const playersWriting: Player[] = [];
```

Comment:
ES6 Set could be used here
```suggestion
const playersWriting = new Set<Player>();
```